nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -275,7 +290,16 @@ def forward(
             encoder_input = None
 
         # Call forward on GPT model with preprocessed embeddings
-        if self.autocast_dtype == torch.float32:
+        if self.frozen_model.mcore_gpt:
+            output = self.frozen_model.model(
+                input_ids=None,
+                position_ids=None,
+                decoder_input=encoder_input,
+                attention_mask=attention_mask,
+                labels=labels,
+                inference_params=inference_params,
+            )
+        elif self.autocast_dtype == torch.float32:
```

Comment:
What does autocast_dtype have to do with this? with and without it looks like identical calls to me?

Reply:
yes. I'm wondering the same thing here. this is not my code. I only added the mcore if statement. Should we combine the other 2 cases?

Reply:
Yes let's delete the redundant conditional and just check if mcore or not

Reply:
done.